src/sentry/api/endpoints/project_rule_details.py

```diff
@@ -285,7 +285,9 @@ def put(self, request: Request, project, rule) -> Response:
                 context = {"uuid": client.uuid}
                 return Response(context, status=202)
 
-            trigger_sentry_app_action_creators_for_issues(actions=kwargs["actions"])
+            result = create_sentry_app_alert_rule_issues_component(actions=kwargs["actions"])
```

Comment:
[2] Then wrap this in try catch and then in the catch, catch the exception and call a method like `return IntegrationResponse.from_exception()` where this method is a util that does all the work you're doing to turn exceptions into response.

If from_exception() needs to know the caller for some reason to parse the errors better you can pass that as a param.